interface/patient_file/report/custom_report.php

```diff
@@ -625,27 +667,43 @@ function postToGet($arin)
                         echo "</div></div>\n"; // HTML to PDF conversion will fail if there are open tags.
                         $content = getContent();
                         $pdf->writeHTML($content); // catch up with buffer.
-                        $pg_header = "<span>" . xlt('Document') . " " . text($fname) ."</span>";
-                        $tempDocC = new C_Document;
-                        $pdfTemp = $tempDocC->retrieve_action($d->get_foreign_id(), $document_id, false, true, true, true);
+                        $err = '';
+                        try {
+                            // below header isn't being used. missed maybe!
+                            $pg_header = "<span>" . xlt('Document') . " " . text($fname) ."</span>";
+                            $tempDocC = new C_Document;
+                            $pdfTemp = $tempDocC->retrieve_action($d->get_foreign_id(), $document_id, false, true, true, true);
                         // tmp file in temporary_files_dir
-                        $from_file_tmp_name = tempnam($GLOBALS['temporary_files_dir'], "oer");
-                        file_put_contents($from_file_tmp_name, $pdfTemp);
-                        $pagecount = $pdf->setSourceFile($from_file_tmp_name);
-                        for ($i = 0; $i < $pagecount; ++$i) {
-                            $pdf->AddPage();
-                            $itpl = $pdf->importPage($i+1);
-                            $pdf->useTemplate($itpl);
-                        }
-                        unlink($from_file_tmp_name);
-
-                        // Make sure whatever follows is on a new page.
-                        $pdf->AddPage();
-
+                            $from_file_tmp_name = tempnam($GLOBALS['temporary_files_dir'], "oer");
+                            file_put_contents($from_file_tmp_name, $pdfTemp);
+                            $pagecount = $pdf->setSourceFile($from_file_tmp_name);
+                            for ($i = 0; $i < $pagecount; ++$i) {
+                                $pdf->AddPage();
+                                $itpl = $pdf->importPage($i+1);
+                                $pdf->useTemplate($itpl);
+                            }
+                        } catch (Exception $e) {
+                            // chances are PDF is > v1.4 and compression level not supported.
+                            // regardless, we're here so lets dispose in different way.
+                            //
+                            unlink($from_file_tmp_name);
+                            $archive_name = ($GLOBALS['temporary_files_dir'] . '/' . report_basename($pid)['base'] . ".zip");
+                            $rtn = zip_content(basename($d->url), $archive_name, $pdfTemp);
```

Comment:
Gonna likely have 2 potential problems with this:
1. $d->url is not reliable way to find the file; bit more complicated as is done here: https://github.com/openemr/openemr/blob/master/controllers/C_Document.class.php#L671-L687
2. The file is encrypted

Would it make sense to drop the first parameter in zip_content and just use the source?

Reply:
I meant to say content, but I may not fully understand what is going in that function.

Reply:
two options for content in zip_content. by file or content. note I'm using url to get file name not retrieve content in this case. wrote this this way in case I wanted to archive by retrieving the temp file content by file or etc...take closer look at zip_content routine.